### ISSUE TITLE
How to set up multiple GPUs for model performance verification on the val set during YOLOV8 training process

### ISSUE THREAD

**Issue by litengfeigithub** [ultralytics/ultralytics#16335]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

During the training process, I set it to use 4 GPUs, and I validate the validation set after each epoch. However, I noticed that it seems to use only one GPU for validation instead of four. Where can I set this parameter?

**Comment by Y-T-G** [comment]
Unfortunately, multi-GPU validation is not supported currently. It only runs on a single GPU.

**Comment by Haseeeb21** [comment]
@litengfeigithub Hey there,
Well you can train model on different GPU's by adding, device = (0,1) in train as parameter.

`results = model.train(data="data.yaml", epochs=100, imgsz=640, device = (0,1))`

Now the model will be training on both the GPU's.

**Comment by DoubleFliker** [comment]
When will this function be online? The multi-GPUs validation during training will accelerate the training process faster. ^ ^

**Comment by glenn-jocher** [comment]
@DoubleFliker thank you for the suggestion! Multi-GPU validation during training is not currently supported but is a great idea for future improvements. Please stay updated by following the [Ultralytics repository](https://github.com/ultralytics/ultralytics) for any new feature announcements.

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in the pull request Enable multi-GPU validation during training [ultralytics#22377](https://github.com/ultralytics/ultralytics/pull/22377) by @Y-T-G with contributions from @glenn-jocher and @ExtReMLapin. 

Key changes that may resolve the issue:
- Added a ContiguousDistributedSampler for non-shuffled DDP DataLoaders to keep batch-aligned, contiguous indices per GPU, improving padding efficiency and consistency.
- Made validation DDP-aware: each rank builds its own val DataLoader/Validator, EMA buffers are broadcast from rank 0, and metrics/losses/plots are reduced and reported only on rank 0.
- Implemented robust stats gathering for classification and detection, ensuring accurate aggregation on rank 0.
- Adjusted final evaluation flow to work correctly under DDP and validate best.pt cleanly.
- Version bumped to 8.3.218.

How to test the fix:
- Upgrade to the latest version with the changes:
  - `pip install -U "ultralytics>=8.3.218"`
- Or install from the latest main branch:
  - `pip install -U "ultralytics @ git+https://github.com/ultralytics/ultralytics.git@main"`
- Optional quick multi-GPU check (example):
  - `yolo detect train model=YOLO11n.pt data=coco8.yaml devices=0,1`

Feedback requested:
- Please try the updated version and confirm whether multi-GPU validation during training now runs as expected and metrics/logging are consistent.

Thanks for reporting this, and feel free to share logs or a minimal repro if anything persists. 🙏
